### ISSUE TITLE
beryllium: camera doesn't work

### ISSUE THREAD

**Issue by entrider** [Divested-Mobile/DivestOS-Build#37]
As specified on [this page](https://divestos.org/index.php?page=broken), camera doesnt work on my Pocophone F1. As I figured out, this issue is probably related to removing some DRM blobs in DivestOS. Because when I installed a vanilla LineageOS 18.1 camera worked fine, but after i applied the [Magisk DRM Disabler module](https://github.com/Magisk-Modules-Repo/magisk-drm-disabler), it stopped working. I can investigate further if needed. 
Would it be possible to whitelist some of required blobs to make camera work on DivestOS?

**Comment by SkewedZeppelin** [comment]
Yes, of course.

Just still haven't figured out which is the culprit.

**Comment by dimqua** [comment]
To make OpenCamera work, it seems that either these should be whitelisted:
https://github.com/Divested-Mobile/DivestOS-Build/blob/dbd2a717228b04660c2b0ece6cdf86f0f0f0d2e9/Patches/Common/android_vendor_divested/firmware_deblobber/firmware_deblobber.sh#L27-L32
or these:
https://github.com/Divested-Mobile/DivestOS-Build/blob/dbd2a717228b04660c2b0ece6cdf86f0f0f0d2e9/Patches/Common/android_vendor_divested/firmware_deblobber/firmware_deblobber.sh#L34-L42
I'd suggest to whitelist the first set, see if there will be any stability issues, if yes, it may be needed to whitelist the other set instead or both of them.

However, even when both of these sets of blobs are present Google Camera refuses to work.

**Comment by entrider** [comment]
To make OpenCamera work, it seems that either these should be whitelisted:
https://github.com/Divested-Mobile/DivestOS-Build/blob/dbd2a717228b04660c2b0ece6cdf86f0f0f0d2e9/Patches/Common/android_vendor_divested/firmware_deblobber/firmware_deblobber.sh#L27-L32
or these:
https://github.com/Divested-Mobile/DivestOS-Build/blob/dbd2a717228b04660c2b0ece6cdf86f0f0f0d2e9/Patches/Common/android_vendor_divested/firmware_deblobber/firmware_deblobber.sh#L34-L42
I'd suggest to whitelist the first set, see if there will be any stability issues, if yes, it may be needed to whitelist the other set instead or both of them.

However, even when both of these sets of blobs are present Google Camera refuses to work.

**Comment by SkewedZeppelin** [comment]
That script you linked isn't used in years.

**Comment by SkewedZeppelin** [comment]
Please run on a fresh boot:

adb logcat -b all -d | grep -i -e dlopen -e gyro -e accel -e sensor -e camera -e
audit

**Comment by entrider** [comment]
> Please run on a fresh boot: adb logcat -b all -d | grep -i -e dlopen -e gyro -e accel -e sensor -e camera -e audit

https://paste.debian.net/plainh/91407c1f

**Comment by SkewedZeppelin** [comment]
Is this still an issue on the 20220213 build?

**Comment by ushilives** [comment]
> Is this still an issue on the 20220213 build?

Just tested out the 20220301 build and the camera still doesn't work. As a temporary solution can i flash Lineage OS 18.1 Vendor on top of Divest OS?

**Comment by SkewedZeppelin** [comment]
You can try

**Comment by ushilives** [comment]
> You can try

Just tried it, still doesn't work :-(

**Comment by ushilives** [comment]
> Reflash DivestOS to get the vendor image back. Then try flashing the boot image from latest Lineage. If that fixes it, it'll narrow things down.

It works! Clean flashed DivestOS then installed boot.img from Lineage OS 18.1 as you suggested.

**Comment by SkewedZeppelin** [comment]
You lose out on the defconfig hardening, the hardening patches, and 31 security fixes.

But I make 3 camera related fixes to the kernel to beryllium recently https://github.com/Divested-Mobile/DivestOS-Build/commit/ac1e89f0c89de4c6490f6824c5e5e22a42433e5b#diff-6fdb4abb2f97f004f3cfecdc4dd1dfcf1d70700a4378269399f5f8878e32bacd, so hopefully next build works.

**Comment by ushilives** [comment]
> You lose out on the defconfig hardening, the hardening patches, and 31 security fixes.
> 
> But I did make 3 camera related fixes to the kernel to beryllium recently [ac1e89f#diff-6fdb4abb2f97f004f3cfecdc4dd1dfcf1d70700a4378269399f5f8878e32bacd](https://github.com/Divested-Mobile/DivestOS-Build/commit/ac1e89f0c89de4c6490f6824c5e5e22a42433e5b#diff-6fdb4abb2f97f004f3cfecdc4dd1dfcf1d70700a4378269399f5f8878e32bacd), so hopefully next build works.

OK thanks will test out next build when released 👍

**Comment by SkewedZeppelin** [comment]
A new update is now up with 3 camera related fixes for beryllium, please test.

**Comment by ushilives** [comment]
> A new update is now up with 3 camera related fixes for beryllium, please test.

Yay! OK will have a test in next few days and will report back to you

**Comment by ushilives** [comment]
I have just tested it and unfortunately it does the same thing. Just freezes on camera and restarts.

**Comment by SkewedZeppelin** [comment]
Thank you for testing, I will have to dig deeper.

**Comment by ushilives** [comment]
Anytime, any new builds I can test. 🤔 for this ROM I habebt got to flash official Miui firmware or any prerequisites beforehand? I randomly tried a different camera app like graphene just in case but still same effect.

**Comment by SkewedZeppelin** [comment]
@ushilives 
No you are fine. It is something that the DivestOS changes break, you confirmed that when you tested the LineageOS boot.img.

**Comment by gagzzz** [comment]
Hello,
The issue is still there, camera is still not working!
Is there a plan to fix it at all?

thank you SO MUCH for all the work you are doing, it's really amazing!

**Comment by Viktor-11** [comment]
Hello everybody,

Camera is still not working yet!
will anybody fix the bug ?

Thank you for good Rom , but we have only one bug...
pls fix it...

**Comment by Viktor-11** [comment]
in last year we have that bug too...

**Comment by werdahias** [comment]
Hi, is still issue still present ? I'd like to know before I format my current Lineage install

**Comment by SkewedZeppelin** [comment]
@werdahias
yes, it is still an issue: https://divestos.org/pages/broken#beryllium

**Comment by werdahias** [comment]
thanks, anything I can do to help debugging this ?

**Comment by SkewedZeppelin** [comment]
@werdahias 
someone needs to bisect every change in https://github.com/Divested-Mobile/DivestOS-Build/blob/master/Scripts/LineageOS-20.0/CVE_Patchers/android_kernel_xiaomi_sdm845.sh and https://github.com/Divested-Mobile/DivestOS-Build/blob/master/Scripts/Common/Functions.sh#L905-L1166

**Comment by werdahias** [comment]
so I'd need to build every image with said patched / kconfigs applied / unapplied seperately and test those ?

**Comment by SkewedZeppelin** [comment]
yes, you'd likely have to compile the kernel 10-30 times and test each one to narrow it down

would take like 1-5 hours

I've already put probably two dozen hours into trying to fix this issue over the years, but I don't have one of these devices

**Comment by vm-project-repo** [comment]
Hi everybody,

is still issue still present ?
